pkg/kubelet/cm/cpumanager/policy_static.go

```diff
@@ -209,6 +209,17 @@ func (p *staticPolicy) Allocate(s state.State, pod *v1.Pod, container *v1.Contai
 			return err
 		}
 		s.SetCPUSet(string(pod.UID), container.Name, cpuset)
+
+		// Check if the container that has just been allocated resources is an init container.
+		// If so, release its CPUs back into the shared pool so they can be reallocated.
+		for _, initContainer := range pod.Spec.InitContainers {
```

Comment:
this makes sense per the prior commit to ensure that init containers have cpusets that are exclusive only to life of the init container.

Reply:
each handler runs in sequence:

https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/kubelet.go#L1793

all pod admission is handled in HandlePodAdditions call
https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/kubelet.go#L2070

the sync loop iteration is not asynch which is where all pod additions are called out from:
https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/kubelet.go#L1894